unstructured/ingest/connector/azure_cognitive_search.py

```diff
@@ -0,0 +1,114 @@
+import json
+import typing as t
+from dataclasses import dataclass
+
+import azure.core.exceptions
+
+from unstructured.ingest.error import WriteError
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class SimpleAzureCognitiveSearchStorageConfig(BaseConnectorConfig):
+    endpoint: str
+    key: str
+
+
+@dataclass
+class AzureCognitiveSearchWriteConfig(WriteConfig):
+    index: str
+
+
+@dataclass
+class AzureCognitiveSearchDestinationConnector(BaseDestinationConnector):
+    write_config: AzureCognitiveSearchWriteConfig
+    connector_config: SimpleAzureCognitiveSearchStorageConfig
+
+    @requires_dependencies(["azure"], extras="azure-cognitive-search")
+    def initialize(self):
+        from azure.core.credentials import AzureKeyCredential
+        from azure.search.documents import SearchClient
+
+        # Create a client
+        credential = AzureKeyCredential(self.connector_config.key)
+        self.client = SearchClient(
+            endpoint=self.connector_config.endpoint,
+            index_name=self.write_config.index,
+            credential=credential,
+        )
+
+    def conform_dict(self, data: dict) -> None:
```

Comment:
should this be a common function that can be reused by other connectors that need to do this?

Reply:
also maybe quick docstring just explaining what/why we're doing this

Reply:
At least right now this is specific to the schema used by Azure Cognitive Search index, but I could see something similar being used later for other json-based doc stores. Will add doc string but leave this in here for now.